code/game/gamemodes/meteor/meteors.dm

```diff
@@ -308,12 +312,29 @@
 /obj/effect/meteor/meaty
 	name = "meaty ore"
 	icon_state = "meateor"
-	meteordrop = /obj/item/reagent_containers/food/snacks/meat/monkey
+	meteor_loot = list(/obj/item/reagent_containers/food/snacks/meat/monkey)
 	dropamt = 10
 
 /obj/effect/meteor/meaty/meteor_effect()
 	gibs(loc)
 
+/obj/effect/meteor/ship_debris
+	name = "ship debris"
+	icon_state = "dust"
+	meteor_loot = list(/obj/item/stack/material/plasteel, /obj/item/stack/material/steel, /obj/item/material/shard, /obj/item/material/shard/shrapnel, /obj/structure/closet/crate/loot)
+	dropamt = 10
+
+/obj/effect/meteor/ship_debris/meteor_effect()
+	for(var/eligible_turf in RANGE_TURFS(2, src))
+		if(prob(25))
+			new /obj/effect/gibspawner/robot(eligible_turf)
+		if(prob(10))
+			new /obj/effect/gibspawner/human(eligible_turf)
+			new /obj/random/voidsuit/no_nanotrasen(eligible_turf)
+		if(prob(65))
+			continue
```

Comment:
Use else if / else here or just remove the last two lines

Reply:
i want all of these to have a chance of occuring on a tile, pushed the updated code tho